frontend-react/src/components/FileHandlers/FileHandler.tsx

```diff
@@ -91,25 +106,54 @@ export function getClientHeader(
 
 function FileHandler() {
     const { state, dispatch } = useFileHandler();
-    const [fileContent, setFileContent] = useState("");
-
     const {
         fileInputResetValue,
         contentType,
         fileType,
         fileName,
         errors,
+        errorType,
         destinations,
         reportId,
         successTimestamp,
-        cancellable,
-        errorType,
         warnings,
         localError,
-        overallStatus,
         reportItems,
         selectedSchemaOption,
     } = state;
+    const [fileContent, setFileContent] = useState("");
+    const [fileHandlerStepIndex, setFileHandlerStepIndex] = useState(0);
```

Comment:
The index is something that should be computed rather than set explicitly; explicitly setting the numerical value for the index technically allows for a user to get to a point where we're making incorrect assumptions about their current state.

Basically the index equals (building on assumptions each step of the way):
- 0 if there's no schema set 
- 1 if they have not submitted anew (and they have no file, but that's implied)
- 2 if they have submitted and there are warnings/errors
- 3 if they have submitted and 1) there are warnings and they've confirmed going to the next step OR 2) there are nor warnings/errors

Technically, we could break this out into a function on its own with the props/reducer state passed in as the parameters.  That's kind of what the [reducer in UseFilerHandler.ts](https://github.com/CDCgov/prime-reportstream/blob/master/frontend-react/src/hooks/UseFileHandler.ts#L214) is getting at: setting programmatic steps based on state, and we can just make an inferred value (the index in this case) from the all the state we have in it.  Happy to chat more about this, though!